workflow-ui/modal-android/src/androidTest/java/com/squareup/workflow1/ui/modal/test/ModalViewContainerLifecycleActivity.kt

```diff
@@ -68,5 +70,29 @@ internal class ModalViewContainerLifecycleActivity : AbstractLifecycleTestActivi
     },
   )
 
+  val currentModalView: ViewStateTestView<LeafRendering>?
+    get() {
+      val modalContainer = rootRenderedView as ModalContainer<*>
+      val firstModalDecor = modalContainer.dialogDecorViews.firstOrNull() ?: return null
+      @Suppress("UNCHECKED_CAST")
+      return (((firstModalDecor as ViewGroup)
+        .getChildAt(0) as ViewGroup)
+        .getChildAt(0) as ViewGroup)
+        .getChildAt(0) as ViewStateTestView<LeafRendering>
```

Comment:
I think this assumption of structure is failing ui tests on certain Android versions:

```

com.squareup.workflow1.ui.modal.test.ModalViewContainerLifecycleTest > modal_hierarchy_state_restored_after_config_change[test(AVD) - 5.0.2] FAILED 
1099
	java.lang.RuntimeException: java.lang.ClassCastException: androidx.appcompat.widget.AppCompatTextView cannot be cast to android.view.ViewGroup
1100
	at androidx.test.runner.MonitoringInstrumentation.runOnMainSync(MonitoringInstrumentation.java:450)
1101

```

Reply:
Fixed